### PR TITLE
fix turb_source bug and set CBL_model similar to SBL_model

### PR DESCRIPTION
**Pull request by yairchn**
### Description
This PR take the changes to convective_bl_model.jl from #1817 that do not effect the output of a test and adds them to master separately. 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: experiments/AtmosLES/convective_bl_model.jl [hunk 1]
```diff
@@ -279,20 +313,26 @@ function convective_bl_model(
                 turbconv = turbconv_bcs(turbconv)[1],
             ),
             AtmosBC(turbconv = turbconv_bcs(turbconv)[2]),
-        ),
+        )
+    end
+
+    moisture_flux = FT(0)
```
Comment (charleskawczynski) [inline experiments/AtmosLES/convective_bl_model.jl:319]: Can we move this up to before where it is used?

**Review by ilopezgp — APPROVED** [review]

File: experiments/AtmosLES/convective_bl_model.jl [hunk 2]
```diff
@@ -279,20 +314,25 @@ function convective_bl_model(
...
+
+    problem = AtmosProblem(
         init_state_prognostic = ics,
+        boundaryconditions = boundary_conditions,
     )
 
     # Assemble model components
     model = AtmosModel{FT}(
-        AtmosLESConfigType,
+        config_type,
```
Comment (charleskawczynski) [inline experiments/AtmosLES/convective_bl_model.jl:327]: ❤️ nice catch!

**Comment by yairchn** [comment]
bors try

**Comment by yairchn** [comment]
bors try

**Pull request merged**